docs/modal_deployment.md

```diff
@@ -0,0 +1,107 @@
+# Deploying your own Sweep instance on modal
+
+## Prerequisites
+Ensure you have the following:
+
+- OpenAI API-key, preferably with access to a 32k context GPT-4 model
+- Network exposed MongoDB and Redis instances
+- Poetry package manager installed on your machine
+- Account on modal.com (free-tier)
+
+> This guide is set for a minimal working setup. Feel free to take a look at `sweepai/config/server.py` and adjust your steps as per requirements.
+
+## Steps
+---
+1. **Install the modal client, link it to your account:**
+
+    ```bash
+    pip install modal-client
+    modal token new
+    ```
+---
+2. **Clone the sweep repository and install dependencies:**
+
+    ```bash
+    git clone git@github.com:sweepai/sweep.git
+    cd sweep
+    poetry install
+    ```
+---
+3. **Register a new GitHub App:**
+
+    Use [this guide](https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/registering-a-github-app) to register a new GitHub App. 
+
+    Permissions:
+    ```
+    actions: read
+    check: read
+    contents: read & write
+    commit statuses: read & write
+    issues: read & write
+    metadata: read
+    pull requests: read & write
+    workflows: read & write
+   ```
+
+   Webhook events:
+
+   ```text
+    issues
+    pull request
+    workflow run
+    check run
+    issue comment
+    label
+    pull request review
+    pull request review comment
+    pull request thread
+    status
+    workflow job
+    create
+    commit comment
+    check suite
+    push
+   ```
+    
+   Disable the webhook for now. You can use any URL for the homepage (e.g. repository URL). [Install the app](https://docs.github.com/en/apps/using-github-apps/installing-your-own-github-app) on your account or organization and generate a private key as per [this guide](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/managing-private-keys-for-github-apps). 
+---
+
+4. **Setup modal environment:**
+
+   - Navigate to the 'Secrets' tab in Modal.
+   - Create a 'github' secret with the following values:
+    ```cmd
+    GITHUB_APP_ID: Your App ID from general tab in Application settings
+    GITHUB_BOT_USERNAME: Your App name
```

Comment:
Don't need username

Reply:
> Don't need username

I see a lot of mentions of it in the code, are you sure?
![image](https://github.com/sweepai/sweep/assets/47783525/0a8d4258-4acb-4527-ab5a-3f03f518f394)

Reply:
Woops you're right. I think we just put the secret somewhere else but that works. Nice catch.